tests/integration/conftest.py

```diff
@@ -695,3 +696,24 @@ async def ingress_application_related_fixture(application: Application, external
         raise_on_error=False,
     )
     return traefik
+
+
+@pytest.fixture(scope="module", name="jenkins_new_job_configuration")
+def jenkins_new_job_configuration() -> str:
+    """XML configuration for blank jenkins job for testing purposes."""
+    return textwrap.dedent(
+        """<project>
+            <description/>
+            <properties/>
+            <scm class="hudson.scm.NullSCM"/>
+            <canRoam>true</canRoam>
+            <disabled>false</disabled>
+            <blockBuildWhenDownstreamBuilding>false</blockBuildWhenDownstreamBuilding>
+            <blockBuildWhenUpstreamBuilding>false</blockBuildWhenUpstreamBuilding>
+            <triggers class="vector"/>
+            <concurrentBuild>false</concurrentBuild>
+            <builders/>
+            <publishers/>
+            <buildWrappers/>
+        </project>"""
+    )
```

Comment:
There's a helper function for generating a job xml in `tests/integration/helpers.py` `gen_test_job_xml`, you can pass in a random node label to it to generate a job xml targeting that particular worker node. I think it would help reduce duplicate code! :D

Reply:
thanks, I updated the test to use the helper instead and remove this duplicate code